android/core/src/main/java/com/stadiamaps/ferrostar/core/NavigationViewModel.kt

```diff
@@ -60,22 +63,22 @@ interface NavigationViewModel {
 class DefaultNavigationViewModel(
     private val ferrostarCore: FerrostarCore,
     private val spokenInstructionObserver: SpokenInstructionObserver? = null,
-    locationProvider: LocationProvider
+    private val locationProvider: LocationProvider
 ) : ViewModel(), NavigationViewModel {
 
-  private var lastLocation: UserLocation? = locationProvider.lastLocation
+  private var snappedLocation: UserLocation? = locationProvider.lastLocation
 
   override val uiState =
       ferrostarCore.state
           .map { coreState ->
-            lastLocation =
+              val location = locationProvider.lastLocation
+            snappedLocation =
                 when (coreState.tripState) {
```

Comment:
I think I didn't specify very well how I envision this being used (my bad!).

I _think_ it would be useful to have a Boolean argument to the default navigation view model which controls whether the snapped location is used (when available).

It would still need to fall back to the unsnapped location when not navigating. Having a booked though to prefer snapped location would allow for as much flexibility as I can think of at the moment. Maybe @Archdoog also has an opinion on the most useful interface.

Reply:
Thinking through the goals here, maybe we should:

1. Add a `rawLocation` to the NavigationState in common. This would generally simplify the stack if a dev wants to use it for some custom behavior (whether UI or other). This is easy to accomplish since we already have single location provider under the hood passing through the `NavigationController`.
2. We could enhance this by providing an encapsulated `fun userLocation(userLocationMode: UserLocationMode) -> UserLocation`. Where `UserLocationMode` is an enum of raw location, snapped location, etc. Then the view model could just publish whatever the location is for the desired mode. This behavior/pattern could also be more useful globally for other platforms - not just android.

Side note, this might be a good point to brainstorm feature bloat on `NavigationState`. I think we're consistently going to have the desire to pack more information into that object (or multiple objects that distill navigation info), so we might want to standardize our process a bit. E.g. separate objects or namespaces like `navigationState.location`, `navigationState.instructions`, etc? 

Thoughts?

Reply:
I feel like adding this to the `NavigationState` is a bad idea for several reasons (and said as much in this comment: https://github.com/stadiamaps/ferrostar/issues/209#issuecomment-2342570373).

* As you said, it is really bloating the state
* It doesn't even have anything to do with navigation. Snapped location is very much specific to navigation, but the user's "raw" location is not in any way tied to `NavigationState`. We want to use this outside the context of a trip; it's not hard to get the data; we just want to expose it in the observed _UI state_.

Reply:
@Archdoog and I discussed this a bit in the core devs meeting today and will play around with it as-is first before making a decision on what changes need to be made (if any; we realized that it might actually be great as-is but that we just need to actually _use_ it first). No action needed on your part right now @cu8code; we'll probably either 1) merge as-is (after fixing lints or w/e), or 2) have a better idea after using it for a bit.

Sorry to hold this up, but I'll revisit this later in the week :)

Reply:
Sorry for the delay on this. Yes, it looks like your existing approach is all we need at the view model level :) I've gone ahead and added the conveniences I was looking for at a higher level (the composable views), which is where they belong anyways.